src/ExtendedTableWidget.cpp

```diff
@@ -305,15 +296,39 @@ void ExtendedTableWidget::copy(const bool withHeaders)
             htmlResult.append(fieldSepHtml);
         }
         currentRow = index.row();
-        QByteArray text = index.data(Qt::EditRole).toByteArray();
 
-        // Table cell data
-        if (text.contains('\n') || text.contains('\t'))
-          htmlResult.append("<pre>" + QString(text).toHtmlEscaped() + "</pre>");
-        else
-          htmlResult.append(QString(text).toHtmlEscaped());
+        QImage img;
+        QVariant data = index.data(Qt::EditRole);
 
-        result.append(escapeCopiedData(text));
+        // Table cell data: image? Store it as an embedded image in HTML and as base 64 in text version
+        if (img.loadFromData(data.toByteArray()))
+        {
+            QByteArray ba;
+            QBuffer buffer(&ba);
+            buffer.open(QIODevice::WriteOnly);
+            img.save(&buffer, "PNG");
+            buffer.close();
+
+            QString imageBase64 = ba.toBase64();
+            htmlResult.append("<img src=\"data:image/png;base64,");
+            htmlResult.append(imageBase64);
+            result.append(imageBase64); // TODO: Or should be just "Image"?
+            htmlResult.append("\" alt=\"Image\">");
+        } else {
+            QByteArray text;
+            if (m->isBinary(index))
+                text = data.toByteArray().toBase64(); // TODO: Or should be just "BLOB"?
```

Comment:
I would probably leave them empty like we did before. So no "Image" and no "BLOB" but no data either if that makes sense. The reason is that probably no one will parse the Base64 when pasting anyway. And because it's impossible to tell if the contents of a cell was actual binary data, "BLOB", or some Base64 string I think it's best to just not copy anything. It's probably the clearest way to tell the user that this isn't going to work as expected :wink:

Reply:
Great! That isn't really very useful and my TODO note was already indicating my hesitation. No objection at all.

Reply:
Cool, just changed it :smile: